frontend/src/components/pages/LogRecords.tsx

```diff
@@ -1,106 +1,281 @@
+  const [tags, setTags] = useState("");
+  const [attentionTo, setAttentionTo] = useState("");
+  const [building, setBuilding] = useState("");
+  const [flagged, setFlagged] = useState(false);
+
+  // Record/page state
   const [logRecords, setLogRecords] = useState<LogRecord[]>([]);
-  console.log(logRecords);
+  const [numRecords, setNumRecords] = useState<number>(0);
+  const [resultsPerPage, setResultsPerPage] = useState<number>(25);
+  const [pageNum, setPageNum] = useState<number>(1);
+
+  const getLogRecords = async (page_number: number) => {
+    const employeeIds = employees
+      ? employees.replaceAll(`"`, "").split(",")
+      : [];
+    const attentionTos = attentionTo
+      ? attentionTo.replaceAll(`"`, "").split(",")
+      : [];
+    const dateRange = startDate && endDate ? [startDate, endDate] : [];
+
+    const data = await commonAPIClient.filterLogRecords(
+      building,
+      employeeIds,
+      attentionTos,
+      dateRange,
+      tags ? [tags] : [],
+      flagged,
+      resultsPerPage,
+      page_number,
+    );
+    setLogRecords(data ? data.logRecords : []);
+    setNumRecords(data ? data.numResults : 0);
+
+    if (data?.numResults === 0) {
+      setPageNum(0);
+    } else {
+      setPageNum(page_number);
+    }
+  };
+
+  useEffect(() => {
+    getLogRecords(1);
+  }, [
+    building,
+    employees,
+    attentionTo,
+    startDate,
+    endDate,
+    tags,
+    flagged,
+    setLogRecords,
+    resultsPerPage,
+  ]);
 
   return (
     <div className="page-container">
       <NavigationBar />
       <div className="records">
         <CreateLog />
-        <SearchAndFilters setLogRecords={setLogRecords} />
-        <TableContainer paddingTop="12px">
-          <Table
-            variant="simple"
-            style={{ minHeight: "400px", verticalAlign: "middle" }}
-          >
-            <Thead className="table-header">
-              <Tr>
-                <Th>Date</Th>
-                <Th>Time</Th>
-                <Th>Resident</Th>
-                <Th>Note</Th>
-                <Th>Employee</Th>
-                <Th>Attn To</Th>
-              </Tr>
-            </Thead>
 
-            <Tbody>
-              {/* TODO: replace mockRecords with logRecords */}
-              {mockRecords.map((record) => {
-                return (
-                  <Tr key={record.id} style={{ verticalAlign: "middle" }}>
-                    <Td width="5%">{record.Date}</Td>
-                    <Td width="5%">{record.Time}</Td>
-                    <Td width="5%">{record.Resident}</Td>
-                    <Td whiteSpace="normal" width="75%">
-                      {record.Note}
-                    </Td>
-                    <Td width="5%">{record.Employee}</Td>
-                    <Td width="5%">{record.Attn_To}</Td>
-                  </Tr>
-                );
-              })}
-            </Tbody>
-          </Table>
-        </TableContainer>
+        {/* Filtering dropdown */}
+        <Card>
```

Comment:
oof, I'm not a big fan of removing SearchAndFilters here, but I think that it's hard to pass the states around to different components. What were the rendering issues you were running into?

Reply:
will do

Reply:
Resolved as per reply in suggestion above ^